drivers/gpio/gpio_mchp_mss.c

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2022 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_mpfs_gpio
+
+#include <errno.h>
+#include <kernel.h>
+#include <device.h>
+#include <soc.h>
+#include <drivers/gpio.h>
+#include <sys/util.h>
+
+#include "gpio_utils.h"
+
+
+#define MSS_GPIO_INPUT_MODE               0x02
+#define MSS_GPIO_OUTPUT_MODE              0x05
+#define MSS_GPIO_INOUT_MODE               0x03
+#define MSS_GPIO_IRQ_LEVEL_HIGH           0x00
+#define MSS_GPIO_IRQ_LEVEL_LOW            0x20
+#define MSS_GPIO_IRQ_EDGE_POSITIVE        0x40
+#define MSS_GPIO_IRQ_EDGE_NEGATIVE        0x60
+#define MSS_GPIO_IRQ_EDGE_BOTH            0x80
+#define MSS_GPIO_INT_ENABLE_MASK          0x08
+#define MSS_OUTPUT_BUFFER_ENABLE_MASK     0x04
+
+struct mss_gpio_t {
+	unsigned int gpio_cfg[32];
+	unsigned int gpio_irq;
+	const unsigned int gpio_in;
+	unsigned int gpio_out;
+	unsigned int gpio_cfg_all;
+	unsigned int gpio_cfg_byte[4];
+	unsigned int gpio_clr_bits;
+	unsigned int gpio_set_bits;
```

Comment:
It would be preferable to use uint32_t's here.

Reply:
HI @cfriedt Thanks for the review, have made the changes and pushed.